codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/endpoints/DefaultEndpointProviderTestGenerator.kt

```diff
@@ -61,6 +61,43 @@ class DefaultEndpointProviderTestGenerator(
     fun render() {
         writer.addImport("*", namespace = "kotlin.test")
         writer.withBlock("public class #L {", "}", CLASS_NAME) {
+            writer.write("")
+            writer.withBlock(
+                "private fun expectEqualEndpoints(expected: #T, actual: #T) {",
+                "}",
+                RuntimeTypes.SmithyClient.Endpoints.Endpoint,
+                RuntimeTypes.SmithyClient.Endpoints.Endpoint,
+            ) {
+                // Make exceptions ONLY for business metrics attributes
+                writer.withBlock(
+                    "if (actual.attributes.contains(#T) || actual.attributes.contains(#T)) {",
+                    "} else { assertEquals(expected, actual) }",
+                    RuntimeTypes.Core.BusinessMetrics.serviceEndpointOverride,
+                    RuntimeTypes.Core.BusinessMetrics.accountIdBasedEndPoint,
+                ) {
+                    writer.write(
+                        "val updatedAttributes = expected.attributes.#T()",
+                        RuntimeTypes.Core.Collections.toMutableAttributes,
+                    )
+                    writer.write(
+                        "if (actual.attributes.contains(#T)) updatedAttributes[#T] = true",
+                        RuntimeTypes.Core.BusinessMetrics.serviceEndpointOverride,
+                        RuntimeTypes.Core.BusinessMetrics.serviceEndpointOverride,
+                    )
+                    writer.write(
+                        "if (actual.attributes.contains(#T)) updatedAttributes[#T] = true",
+                        RuntimeTypes.Core.BusinessMetrics.accountIdBasedEndPoint,
+                        RuntimeTypes.Core.BusinessMetrics.accountIdBasedEndPoint,
+                    )
+                    writer.write(
+                        "val newExpected = #T(expected.uri, expected.headers, updatedAttributes)",
+                        RuntimeTypes.SmithyClient.Endpoints.Endpoint,
+                    )
+                    writer.write("assertEquals(newExpected, actual)")
+                }
```

Comment:
**Correctness**: This appears to be setting the _expected_ attributes based on the _actual_ attributes, which is backwards. If the actual attributes erroneously omitted, for example, account ID then this test wouldn't catch it.